fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java

```diff
@@ -784,6 +784,13 @@ public void validateForCreate(final JsonCommand command) {
         final String loanScheduleType = this.fromApiJsonHelper.extractStringNamed(LoanProductConstants.LOAN_SCHEDULE_TYPE, element);
         baseDataValidator.reset().parameter(LoanProductConstants.LOAN_SCHEDULE_TYPE).value(loanScheduleType).ignoreIfNull()
                 .isOneOfEnumValues(LoanScheduleType.class);
+        if (!LoanScheduleType.PROGRESSIVE.equals(LoanScheduleType.valueOf(loanScheduleType))
```

Comment:
For backward compatibility, the loan schedule type is optional. So it might not be sent, in that case the `CUMULATIVE` is the fallback default value. You might wanna mimic that behaviour here as well.